### ISSUE TITLE
Failed to create MLModel from graph data

### ISSUE THREAD

**Issue by FrankdenUijl** [natml-hub/Hand-Pose#1]
I'm getting the following exception: `ArgumentException: Failed to create MLModel from graph data`

Also I'm unable to go to my profile page on https://hub.natml.ai/explore but the access code should be correct.

![image](https://user-images.githubusercontent.com/8240783/199250201-da4355aa-ac4e-48a5-9d58-3268c06c2f35.png)

**Comment by FrankdenUijl** [comment]
It does run on device, does this mean that I can't run this in Unity Editor?

**Comment by olokobayusuf** [comment]
Hey @FrankdenUijl , first I recommend upgrading to NatML 1.0.19, the latest version. Beyond that, can you share the editor logs in a `.txt` attachment? Those logs will contain more information on why the model was not created. Finally, why can't you access your profile page on the Hub website?

**Comment by FrankdenUijl** [comment]
Updating to NatML 1.0.19 and NatDevice 1.3.0 did not work. 

![image](https://user-images.githubusercontent.com/8240783/199287265-dd8a804c-6194-4dd6-8a8e-92b579a0c4e7.png)

**Comment by FrankdenUijl** [comment]
Was able to go to my profile page by re-login. But now my access code stopped working.. I double checked it, and it is the correct one, it worked before.

![image](https://user-images.githubusercontent.com/8240783/199337248-36052808-c4ea-4089-975f-41e3a4206c24.png)

**Comment by olokobayusuf** [comment]
> I'm unable to go to my profile, billing, ect. see video: https://www.dropbox.com/s/8q1mza2w7ttb4wq/natml.mp4?dl=0

Does this happen in a different browser, like Chrome? And are you in private or incognito browsing mode? On the backend, all your account details look correct, so it's hard to tell what's going on.

> Updating to NatML 1.0.19 and NatDevice 1.3.0 did not work.

It seems that the model isn't running with DirectML. The fix should be simple:
```csharp
var modelData = ...
modelData.computeTarget = MLModelData.ComputeTarget.CPUOnly;
var model = new MLEdgeModel(modelData);
```

> Was able to go to my profile page by re-login. But now my access code stopped working.. I double checked it, and it is the correct one, it worked before.

Is the bundle ID of your app different from what you have configured in Unity's Player Settings?

**Comment by FrankdenUijl** [comment]
> Does this happen in a different browser, like Chrome? And are you in private or incognito browsing mode? On the backend, all your account details look correct, so it's hard to tell what's going on.

After login out and in everything worked as expected.

> It seems that the model isn't running with DirectML. The fix should be simple:

Care to explain what this fix does? Why was this not in the example project? Does this fix also work on Android/iOS devices?

> Is the bundle ID of your app different from what you have configured in Unity's Player Settings?

![image](https://user-images.githubusercontent.com/8240783/199452055-955dc379-b1d7-41b2-b539-8a84fa8765d9.png)
This is the only place where I can change my package name, I never touched it. Do I also need to add that somewhere in the NatML package?

Also:
Is there a way to speed-up the long wait after pressing play in the editor? I have to wait 10-20 seconds before everything starts working, I expect it checks my access code + download the model data?

**Comment by olokobayusuf** [comment]
> After login out and in everything worked as expected.

Hm this is a weird bug then. Thanks for confirming.

> Care to explain what this fix does? Why did it worked on my Android device but not in the Editor? Why was this not in the example project? Does this fix also work on Android/iOS devices?

NatML uses the platform's native machine learning infrastructure to run models. On Windows, NatML uses DirectML (which is where the error is coming from). So the fix here was to disable DirectML on Windows by using the CPU. You can further refine this workaround by checking `if (Application.platform == RuntimePlatform.WindowsEditor)` before setting the `computeTarget`. This way, on other platforms NatML will still leverage the neural accelerators.

> This is the only place where I can change my package name, I never touched it. Do I also need to add that somewhere in the NatML package? Strange that my access key stopped working only on my Android device... (just checked and still no luck).

[Do you mind sending me an email](mailto:yusuf@natml.ai) with the logs from the device? We've gotten two other reports of this issue, so it's highest priority.

> Is there a way to speed-up the long wait after pressing play in the editor? I have to wait 10-20 seconds before everything starts working, I expect it checks my access code + download the model data?

The first time you run, there will be a delay from downloading the model, but after that the model will be cached on your device and will load immediately. Nothing else should take long.

**Comment by FrankdenUijl** [comment]
> NatML uses the platform's native machine learning infrastructure to run models. On Windows, NatML uses DirectML (which is where the error is coming from). So the fix here was to disable DirectML on Windows by using the CPU. You can further refine this workaround by checking if (Application.platform == RuntimePlatform.WindowsEditor) before setting the computeTarget. This way, on other platforms NatML will still leverage the neural accelerators.

Thanks!

> [Do you mind sending me an email](mailto:yusuf@natml.ai) with the logs from the device? We've gotten two other reports of this issue, so it's highest priority.

Can you tell me how to get the logs/ what logs from my device?

**Comment by FrankdenUijl** [comment]
Updating to NatDevice 1.3.1 worked 👍

**Comment by olokobayusuf** [comment]
> Can you tell me how to get the logs/ what logs from my device?

Got the logs, and it's the same issue that should have been fixed in 1.3.1.

> Updating to NatDevice 1.3.1 worked 👍

Perfect! Closing this issue.
